src/Sarif.Viewer.VisualStudio/Tags/SarifLocationTagger.cs

```diff
@@ -0,0 +1,573 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+namespace Microsoft.Sarif.Viewer.Tags
+{
+    using Microsoft.CodeAnalysis.Sarif;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Text;
+    using Microsoft.VisualStudio.Text.Editor;
+    using Microsoft.VisualStudio.Text.Tagging;
+    using Microsoft.VisualStudio.TextManager.Interop;
+    using Microsoft.VisualStudio.Utilities;
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Linq;
+    using System.Threading;
+
+    internal class SarifLocationTagger : ITagger<TextMarkerTag>, ISarifLocationTagger, ITextViewCreationListener, IDisposable
```

Comment:
>SarifLocationTagger [](start = 19, length = 19)

Add a comment that indicates that this class is "per text buffer" hence the need for the static dictionaries.